### PR TITLE
{bio}[foss/2023a] dictys v1.1.0 w/ CUDA 12.1.1

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/319

### CODE REVIEW

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/d/dictys/dictys-1.1.0-foss-2023a-CUDA-12.1.1.eb [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        'checksums': ['f1b22e9932123ccf17eebf19e0953c6e9148f589f93d91b872941a696305c83f'],
+    }),
+    ('pyserial', '3.5', {
+        'modulename': 'serial',
+        'checksums': ['3c77e014170dfffbd816e6ffc205e9842efb10be9f58ec16d3e8675b4925cddb'],
+    }),
+    ('ncclient', '0.6.15', {
+        'checksums': ['6757cb41bc9160dfe47f22f5de8cf2f1adf22f27463fb50453cc415ab96773d8'],
+    }),
+    ('junos-eznc', '2.7.0', {
```
Comment (boegel) [inline easybuild/easyconfigs/d/dictys/dictys-1.1.0-foss-2023a-CUDA-12.1.1.eb:73]: @pavelToman As discussed, let's add a standalone easyconfig for `junos-eznc`, and use the latest version (2.7.1) in which we install their `paramiko` fork as an extra extension, and remove the `pip install` command from their `setup.py`.

I've also opened an issue upstream on this, see https://github.com/Juniper/py-junos-eznc/issues/1318

**Comment by boegel** [comment]
Just so we have a trace for this: an earlier revision of this PR, in which `junos-eznc` 2.7.1 was used, resulted in having `paramiko` being added to the `Python/3.11.3-GCCcore-12.3.0` installation, which in the end leads to failing `pip check` runs with errors like:
```
== 2024-08-13 10:11:49,714 easyblock.py:4292 WARNING build failed (first 300 chars): `/software/Python/3.11.3-GCCcore-12.3.0/bin/python -m pip check` failed:
paramiko 3.4.0 requires bcrypt, which is not installed.
```

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/j/junos-eznc/junos-eznc-2.7.1-GCCcore-12.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }),
+    ('paramiko', '3.4.0', {
+        # Juniper fork of paramiko - compatible with junos-eznc
+        'source_urls': ['https://github.com/Juniper/paramiko/archive/'],
+        'sources': [{'download_filename': 'v%(version)s-JNPR.tar.gz', 'filename': '%(name)s-%(version)s-JNPR.tar.gz'}],
+        'checksums': ['6b3b62e18a2b693169eaa50a7cdd2ab5637fc423205ce85e109cb37722f9eeda'],
+    }),
+    (name, version, {
+        'modulename': 'jnpr.junos',
+        'preinstallopts': "sed -i '6d' setup.py && ",
```
Comment (boegel) [inline easybuild/easyconfigs/j/junos-eznc/junos-eznc-2.7.1-GCCcore-12.3.0.eb:48]: @pavelToman I would make this a bit more explicit to make it clear that we're deleting the line in which `pip install` is used (also to make sure that this line is still removed later if it moves to a different line number):
```suggestion
        'preinstallopts': "sed -i '/pip install/d' setup.py && ",
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3304.joltik.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz, 1 x NVIDIA Tesla V100-SXM2-32GB, 545.23.08, Python 3.6.8
See https://gist.github.com/boegel/3d148718e044a17fb717002dec75f6cc for a full test report.

**Comment by boegel** [comment]
Going in, thanks @pavelToman!

**Pull request merged**